mqtt/mqtt.js

```diff
@@ -72,8 +72,14 @@ nats.subscribe('channel.*', function (msg) {
 
 aedes.authorizePublish = function (client, packet, publish) {
     // Topics are in the form `channels/<channel_id>/messages`
-    var channel = packet.topic.split('/')[1],
-        accessReq = {
+    var channel = /^channels\/(.+?)\/messages$/.exec(packet.topic);
+    if (!channel) {
+        logger.warn('unknown topic');
+        publish(4); // Bad username or password
+        return;
+    }
+    channel = channel[1];
```

Comment:
Please create new variable for this (e.g. `channelId`).

Reply:
done